### PR TITLE
Add helper binaries used in our pipeline to the toolchain package

### PR DESCRIPTION
**Pull request by tduffield**
## Description

These files used to be stored in ci-studio-common, but given work on
that project they now make more sense to be kept in this one.

Signed-off-by: Tom Duffield <tom@chef.io>
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: config/files/helper-bins/install-omnibus-product [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+done
+shift $((OPTIND -1))
+
+# Exit if product_name is not set
+product_name=${product_name:?A product name must be specified}
+
+echo "--- Installing $channel $product_name $product_version"
+download_url="$(mixlib-install download --url --channel "$channel" "$product_name" --version "$product_version")"
+
+mixlib-install install-script | sudo bash -s -- -d "$download_dir" -l "$download_url" 2>&1
```
Comment (scotthain) [inline config/files/helper-bins/install-omnibus-product]: mixlib-install is not part of the toolchain. if we want this to work correctly we need to add it as a base dependency.

Comment (tduffield) [reply config/files/helper-bins/install-omnibus-product]: okay, addressed by adding `omnibus-helper-gems`. Based the list off of what we had in our release engineering repo (for the omnibus ami installation). 

Not sure if there is anything else that needs to be done to make things work.

Comment (jeremiahsnapp) [reply config/files/helper-bins/install-omnibus-product]: @tduffield thanks for installing the gems. Can you please make this `install-omnibus-product` script use something like `../embedded/bin/mixlib-install`? Using the relative path should allow it to work for both `omnibus-toolchain` and `angry-omnibus-toolchain`.

File: config/software/omnibus-helper-bins.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+name "omnibus-helper-bins"
+
+source path: "#{project.files_path}/helper-bins"
+
+license :project_license
+skip_transitive_dependency_licensing true
+
+build do
+  copy "#{project_dir}/*", "#{install_dir}/bin"
```
Comment (jeremiahsnapp) [inline config/software/omnibus-helper-bins.rb]: These should not be copied for Windows platforms.

File: config/software/helper-sh.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+skip_transitive_dependency_licensing true
+
+build do
+
+  %w(
+    check-omnibus-package-signed
+    install-omnibus-product
+  ).each do |helper|
+    erb source: "#{helper}.erb",
+        dest: "#{project_dir}/bin/#{helper}",
```
Comment (jeremiahsnapp) [inline config/software/helper-sh.rb]: All occurrences of `project_dir` should be replaced by `install_dir`.

Reference: https://github.com/chef/omnibus/blob/master/lib/omnibus/software.rb#L549-L590

File: config/software/helper-sh.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        vars: {
+          bin_dir: "#{project_dir}/bin"
```
Comment (jeremiahsnapp) [inline config/software/helper-sh.rb]: You'll need a comma to separate these variables.

**Comment by jeremiahsnapp** [comment]
I added a commit that:

* fixes some syntax
* refactors the erb variables
* adds load-omnibus-toolchain script
* adds equivalent powershell scripts

**Comment by tduffield** [comment]
@jeremiahsnapp it won't let me approve since I opened it, but I like your changes!

**Pull request merged**